eqcorrscan/core/template_gen.py

```diff
@@ -808,13 +811,17 @@ def _template_gen(picks, st, length, swin='all', prepick=0.05,
         if not used_tr:
             Logger.warning('No pick for {0}'.format(tr.id))
     if plot and len(st1) > 0:
+        if not os.path.isdir(plotdir):
+            os.makedirs(plotdir)
         fig1 = tplot(st1, background=stplot, picks=picks_copy,
                      title='Template for ' + str(st1[0].stats.starttime),
                      show=False, return_figure=True)
         fig2 = noise_plot(
             signal=st1, noise=noise, show=False, return_figure=True)
-        fig1.savefig("{0}_template.png".format(st1[0].stats.starttime))
-        fig2.savefig("{0}_noise.png".format(st1[0].stats.starttime))
+        fig1.savefig("{0}/{1}_template.png".format(plotdir,
```

Comment:
Can we use the `save` and `savefile` arguments on these functions as well rather than doing the saving here?